tsl/src/continuous_aggs/common.c

```diff
@@ -1060,9 +1060,14 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 
 			if (!both_buckets_are_equal)
 			{
-				char *offset = DatumGetCString(DirectFunctionCall1(interval_out, offset_datum));
+				char *offset =
+					DatumGetPointer(offset_datum) != NULL ?
+						DatumGetCString(DirectFunctionCall1(interval_out, offset_datum)) :
+						"NULL";
```

Comment:
```suggestion
				char *offset =
					bucket_info.bf->bucket_time_offset != NULL ?
						DatumGetCString(DirectFunctionCall1(interval_out, IntervalPGetDatum(bucket_info.bf->bucket_time_offset))) :
						"NULL";
```
I think there's no need to define either `datum_offset`  and `datum_offset_parent`, just do something like above is enought IMHO.

Reply:
Hmmm it won't work since it's used on line 1057 to check if the offsets are equal.